modules/thrift/include/muzic/uzlib.h

```diff
@@ -0,0 +1,176 @@
+/* SPDX-License-Identifier: Zlib */
+/*
+ * uzlib  -  tiny deflate/inflate library (deflate, gzip, zlib)
+ *
+ * Copyright (c) 2003 by Joergen Ibsen / Jibz
+ * All Rights Reserved
+ * http://www.ibsensoftware.com/
+ *
+ * Copyright (c) 2014-2018 by Paul Sokolovsky
+ *
+ * This software is provided 'as-is', without any express
+ * or implied warranty.  In no event will the authors be
+ * held liable for any damages arising from the use of
+ * this software.
+ *
+ * Permission is granted to anyone to use this software
+ * for any purpose, including commercial applications,
+ * and to alter it and redistribute it freely, subject to
+ * the following restrictions:
+ *
+ * 1. The origin of this software must not be
+ *    misrepresented; you must not claim that you
+ *    wrote the original software. If you use this
+ *    software in a product, an acknowledgment in
+ *    the product documentation would be appreciated
+ *    but is not required.
+ *
+ * 2. Altered source versions must be plainly marked
+ *    as such, and must not be misrepresented as
+ *    being the original software.
+ *
+ * 3. This notice may not be removed or altered from
+ *    any source distribution.
+ */
+
+#ifndef UZLIB_H_INCLUDED
+#define UZLIB_H_INCLUDED
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#ifdef __cplusplus
```

Comment:
I do wonder if this would make more sense to move out of here and into a module, deflate compression/decompression might be useful to non-thrift applications, there is lz4 already

Reply:
Here it's an implementation detail - hacked together from off pieces.

However, I would suggest that we properly design a compression / decompression subsystem that supports multiple different compression formats and then have other code adopt that.

For example, we should look at supporting zip, gzip, bz2, xz. But then also, if any of these require hash tables, then using some common API for that would make sense too.